synapse/rest/admin/experimental_features.py

```diff
@@ -0,0 +1,147 @@
+        self.is_mine = hs.is_mine
+        self.handler = hs.get_experimental_features_manager()
+
+    async def on_GET(
+        self,
+        request: SynapseRequest,
+        user_id: str,
+        feature: str,
+    ) -> Tuple[int, JsonDict]:
+        """
+        Checks if a given feature is enabled for a given user
+        """
+        await assert_requester_is_admin(self.auth, request)
+
+        target_user = UserID.from_string(user_id)
+        if not self.is_mine(target_user):
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST,
+                "User must be local to check what experimental features are enabled.",
+            )
+
+        # do a basic validation of the given feature
+        validated = feature in [
+            "msc3026",
+            "msc2654",
+            "msc3881",
+            "msc3967",
+        ]
+
+        if not validated:
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST, "Please provide a valid experimental feature."
+            )
+
+        enabled = await self.handler.get_feature_enabled(user_id, feature)
+
+        return HTTPStatus.OK, {"user": user_id, "feature": feature, "enabled": enabled}
+
+    async def on_PUT(
+        self, request: SynapseRequest, user_id: str, feature: str
+    ) -> Tuple[int, JsonDict]:
+        """
+        Enables a given feature for the requester
+        """
+        await assert_requester_is_admin(self.auth, request)
+
+        target_user = UserID.from_string(user_id)
+        if not self.is_mine(target_user):
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST,
+                "User must be local to enable experimental features.",
+            )
+
+        # validate the feature
+        validated = feature in [
+            "msc3026",
+            "msc2654",
+            "msc3881",
+            "msc3967",
+        ]
+
+        if not validated:
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST, "Please provide a valid experimental feature."
+            )
+
+        user, feature, enabled = await self.handler.set_feature_for_user(
+            user_id, feature, True
+        )
+
+        return HTTPStatus.OK, {"user": user, "feature": feature, "enabled": enabled}
+
+    async def on_DELETE(
+        self, request: SynapseRequest, user_id: str, feature: str
+    ) -> Tuple[int, JsonDict]:
+        """
+        Disables the requested feature for the given user
+        """
+        await assert_requester_is_admin(self.auth, request)
+
+        target_user = UserID.from_string(user_id)
+        if not self.is_mine(target_user):
+            raise SynapseError(
+                HTTPStatus.BAD_REQUEST,
+                "User must be local to disable an experimental feature.",
+            )
+
+        # validate the feature
+        validated = feature in [
+            "msc3026",
+            "msc2654",
+            "msc3881",
+            "msc3967",
+        ]
```

Comment:
This feels like we should use an enum!

Reply:
I've added an enum, hopefully it was what you were talking about?